Userland/Libraries/LibJS/Bytecode/Op.cpp

```diff
@@ -132,19 +133,41 @@ void NewArray::execute_impl(Bytecode::Interpreter& interpreter) const
     interpreter.accumulator() = Array::create_from(interpreter.global_object(), elements);
 }
 
+// FIXME: Since the accumulator is a Value, we store an object there and have to convert back and forth between that an Iterator records. Not great.
+// Make sure to put this into the accumulator before the iterator object disappears from the stack to prevent the members from being GC'd.
+static Object* iterator_to_object(GlobalObject& global_object, Iterator iterator)
+{
+    auto& vm = global_object.vm();
+    auto* object = Object::create(global_object, nullptr);
+    object->define_direct_property(vm.names.iterator, iterator.iterator, 0);
+    object->define_direct_property(vm.names.next, iterator.next_method, 0);
+    object->define_direct_property(vm.names.done, Value(iterator.done), 0);
+    return object;
+}
+
+static Iterator object_to_iterator(GlobalObject& global_object, Object& object)
+{
+    auto& vm = global_object.vm();
+    return Iterator {
+        .iterator = &MUST(object.get(vm.names.iterator)).as_object(),
```

Comment:
Can't this blow up?
e.g. a user passing in `{ iterator: "a small window into your soul...or something", next: ..., done: false }`

Reply:
No, this is an object we created, control & hopefully never expose to the user.

Reply:
E.g. from `generate_array_binding_pattern_bytecode()`:

```cpp
auto iterator_reg = generator.allocate_register();
generator.emit<Bytecode::Op::Load>(value_reg);
generator.emit<Bytecode::Op::GetIterator>();
generator.emit<Bytecode::Op::Store>(iterator_reg);
...
generator.emit<Bytecode::Op::Load>(iterator_reg);
generator.emit<Bytecode::Op::IteratorToArray>();
```

- In `GetIterator`+`Store` we get an `Iterator`, convert it to an object using the mechanism above, and put that in the accumulator.
- In `Load`+`IteratorToArray`, we load that same object from the accumulator and turn it back into an `Iterator` struct for further processing by the regular AOs.